rclpy/src/rclpy/client.hpp

```diff
@@ -17,67 +17,91 @@
 
 #include <pybind11/pybind11.h>
 
-#include <string>
+#include <rcl/client.h>
+
+#include <memory>
+
+#include "destroyable.hpp"
+#include "handle.hpp"
 
 namespace py = pybind11;
 
 namespace rclpy
 {
-/// Create a client
-/**
- * This function will create a client for the given service name.
- * This client will use the typesupport defined in the service module
- * provided as pysrv_type to send messages over the wire.
- *
- * On a successful call a Capsule pointing to the pointer of the created
- * rclpy_client_t * is returned.
- *
- * Raises ValueError if the capsules are not the correct types
- * Raises RuntimeError if the client could not be created
- *
- * \param[in] pynode Capsule pointing to the node to add the client to
- * \param[in] pysrv_type Service module associated with the client
- * \param[in] service_name Python object containing the service name
- * \param[in] pyqos_profile QoSProfile Python object for this client
- * \return capsule containing the rclpy_client_t
- */
-py::capsule
-client_create(
-  py::capsule pynode, py::object pysrv_type, std::string service_name,
-  py::capsule pyqos_profile);
+class Client : public Destroyable
+{
+public:
+  /// Create a client
+  /**
+   * This function will create a client for the given service name.
+   * This client will use the typesupport defined in the service module
+   * provided as pysrv_type to send messages over the wire.
```

Comment:
@sloretz nit^N: I'd drop `over the wire`:
```suggestion
   * provided as pysrv_type to send messages.
```
We're careful elsewhere not to give details about the transport. Perhaps we should be everywhere.

Reply:
Removed mention of transport in 8e20909efb0dcd2e31bf892c0ccdf236018fa250